### PR TITLE
Use record timestamps for record timestamp tracker in sonar

### PR DESCRIPTION
**Pull request by ericvolp12**

### CODE REVIEW

File: sonar/sonar.go [hunk 1]
```diff
@@ -183,14 +185,15 @@ func (s *Sonar) HandleRepoCommit(ctx context.Context, evt *comatproto.SyncSubscr
 	}
 
 	// Parse time from the event time string
-	t, err := time.Parse(time.RFC3339, evt.Time)
+	evtCreatedAt, err := time.Parse(time.RFC3339, evt.Time)
```
Comment (whyrusleeping) [inline sonar/sonar.go:188]: does RFC3339 work here? I wrote `util.ParseTimestamp` to deal with the large variety of different formats ive seen on records

Comment (whyrusleeping) [reply sonar/sonar.go:188]: Oh, I see you know about that. carry on

Comment (ericvolp12) [reply sonar/sonar.go:188]: Oh I started using a better thing like here: https://github.com/ericvolp12/bsky-experiments/blob/main/pkg/consumer/consumer.go#L256

https://github.com/araddon/dateparse/ - Faster than what we're currently doing i.e. "shotgun parsing"

For Sonar I think it makes sense for me to update to use that library so I'll patch that today.

File: util/time.go [hunk 2]
```diff
@@ -9,6 +9,8 @@ const ISO8601 = "2006-01-02T15:04:05.000Z"
 
 const ISO8601_milli = "2006-01-02T15:04:05.000000Z"
 
+const ISO8601_milli_without_z = "2006-01-02T15:04:05.000000"
```
Comment (whyrusleeping) [inline util/time.go:12]: This isnt valid according to the spec, timestamp is required

**Pull request merged**